packages/core/src/services/documentLogs/computeDocumentLogs.ts

```diff
@@ -0,0 +1,88 @@
+import { and, desc, eq, isNull, sql } from 'drizzle-orm'
+
+import { Commit, DEFAULT_PAGINATION_SIZE, ErrorableEntity } from '../../browser'
+import { database } from '../../client'
+import { calculateOffset } from '../../lib/pagination/calculateOffset'
+import { DocumentLogsRepository } from '../../repositories'
+import {
+  commits,
+  documentLogs,
+  projects,
+  runErrors,
+  workspaces,
+} from '../../schema'
+import { getCommonQueryConditions } from './computeDocumentLogsWithMetadata'
+
+export function computeDocumentLogsQuery(
+  {
+    workspaceId,
+    documentUuid,
+    draft,
+    allowAnyDraft,
+    page = '1',
+    pageSize = String(DEFAULT_PAGINATION_SIZE),
+  }: {
+    workspaceId: number
+    documentUuid?: string
+    draft?: Commit
+    allowAnyDraft?: boolean
+    page?: string
+    pageSize?: string
+  },
+  db = database,
+) {
+  const repo = new DocumentLogsRepository(workspaceId, db)
+  const offset = calculateOffset(page, pageSize)
+  const scope = repo.scope.as('document_logs')
+  const fields = scope._.selectedFields
+  return db
+    .select(fields)
+    .from(scope)
```

Comment:
nein, forbidden, this will fetch all document logs for that workspace before doing any other operation, you have to use the dynamic query or craft your own.

Reply:
Yeah, this sucks a bit no?